android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -264,6 +233,65 @@ public void setBackgroundColor(int colorInt)
 		this.mBottomNavigationView.setBackgroundColor(colorInt);
 	}
 
+	@Override
+	public void updateTabBackgroundDrawable(int index)
```

Comment:
It is ignoring the index because we use the method whenever a new item is added in the tab controller and BottomNavigationMenuView is rebuilding the whole view when this happens. Thus everything not set through the "proper" APIs is reset. This is why we ignore it when setting the background drawable and the text color state list - the component does not provide a direct way to set those properties so I am relying on going through the layout and manually set the new values.

But it will be better to stick to going through all the items only when a new item is added, so I will move this logic in a new private method dealing with new items and respect the index value for other updates.

Reply:
Okay. In this case, this is where a code comment would be helpful. Thanks.

Reply:
Oh never mind. I can see you've already written a code comment about this.